packages/web-app-files/src/composables/viewMode/useViewMode.ts

```diff
@@ -0,0 +1,15 @@
+import { computed, ComputedRef, unref } from 'vue'
+import { useRouteQueryPersisted } from 'web-pkg/src/composables'
+import { ViewModeConstants } from './constants'
+
+export function useViewMode<T>(options: ComputedRef<string>): ComputedRef<string> {
+  if (options) {
+    return computed(() => unref(options))
+  }
+
+  const perPageQuery = useRouteQueryPersisted({
+    name: ViewModeConstants.viewModeQueryName,
+    defaultValue: ViewModeConstants.viewModeDefault
+  })
+  return computed(() => String(unref(perPageQuery)))
```

Comment:
```suggestion
  return computed(() => queryItemAsString(unref(perPageQuery)))
```
+ import from `web-pkg/src/composables`